arestor/api/admin/resource.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2016 Cloudbase Solutions Srl
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+
+"""Arestor API endpoint for resource management."""
+
+import cherrypy
+
+from arestor.api import base as base_api
+from arestor.common import tools as arestor_tools
+from arestor.common import util as arestor_util
+
+# TODO(mmicu): Find a better way to expose this tool
+cherrypy.tools.user_required = arestor_tools.UserManager()
+
+KEY_FORMAT = "{namespace}/{user}/{name}"
+
+
+class ResourceEndpoint(base_api.Resource):
+
+    exposed = True
+
+    """Resource management endpoint."""
+
+    @cherrypy.tools.user_required()
+    @arestor_util.check_credentials
+    @arestor_util.json_response
+    def GET(self, resource_id=None, namespace=None, client_id=None,
+            resource=None):
+        """The representation of userdata resource."""
+        connection = self._redis.rcon
+        response = {"meta": {"status": True, "verbose": "Ok"}, "content": None}
+
+        if resource_id:
+            # Prepare the representation of the received resource.
+            resource = connection.hgetall(resource_id)
+            if not resource:
+                response["meta"]["status"] = False
+                response["meta"]["verbose"] = "Resource not found"
+
+            response["content"] = resource
+            return response
+
+        if not any([resource_id, namespace, client_id, resource]):
+            # Prepare a list with all the available resources.
+            mockdata = connection.keys(pattern="*/*/*")
+            response["content"] = mockdata
+            return response
+
+        namespace = str(namespace) if namespace else "*"
+        client_id = str(client_id) if client_id else "*"
+        resource = str(resource) if resource else "*"
+
+        key = KEY_FORMAT.format(namespace=namespace, user=client_id,
+                                name=resource)
+        mockdata = connection.keys(pattern=key)
+        response["content"] = mockdata
+        return response
+
+    @cherrypy.tools.user_required()
+    @arestor_util.check_credentials
+    @arestor_util.json_response
+    def POST(self, client_id=None, namespace=None, resource=None, **kwargs):
+        """Create a new resource."""
+        connection = self._redis.rcon
+        response = {"meta": {"status": True, "verbose": "Ok"}, "content": None}
```

Comment:
I think a better idea will be to add another layer over the HTTP methods in order to prepare the json response, because this part became redundant.